MapboxNavigation.xcodeproj/project.pbxproj

```diff
@@ -901,7 +901,7 @@
 		C53C19751F38EADE008DB406 /* fr */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = fr; path = fr.lproj/Localizable.strings; sourceTree = "<group>"; };
 		C53C19771F38EAE4008DB406 /* ca */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.strings; name = ca; path = ca.lproj/Localizable.strings; sourceTree = "<group>"; };
 		C53C197A1F38EAEA008DB406 /* Base */ = {isa = PBXFileReference; lastKnownFileType = text.plist.strings; name = Base; path = Base.lproj/Localizable.strings; sourceTree = "<group>"; };
-		C53F2F0720EBC95600D9798F /* Example-CarPlay.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = "Example-CarPlay.app"; sourceTree = BUILT_PRODUCTS_DIR; };
+		C53F2F0720EBC95600D9798F /* Example_CarPlay.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = Example_CarPlay.app; sourceTree = BUILT_PRODUCTS_DIR; };
```

Comment:
Is there a particular reason why the application needed to be renamed from Example-CarPlay to Example_CarPlay? I guess it doesn’t hurt, but it could make it more difficult to merge these changes into the release-v2.0 branch, and we might have other stray references to Example-CarPlay.

Reply:
A couple of things. The UIScene Manifest in the Info.plist file needs an entry for the UISceneDelegate class name. In order for it to match, it needs to contain the product name and the class name, such as "Example.AppDelegate". If the product contains a dash symbol, like Example-CarPlay, the product name in the compiled app will be changed to Example_CarPlay since dashes aren't allowed in the class names. So if I leave our product name as `Example-CarPlay` and try to put an entry in the Info.plist for the UISceneDelegate class name, it will look for `Example_CarPlay.AppDelegate` as the class and fail to find it because the class will actually have a name of `Example-CarPlay.AppDelegate`.

The easiest way to resolve this, short of having a different Info.plist file per target seemed to be to rename the product to Example_CarPlay.

If there are more preferred ways to resolve this I am happy to investigate them.

Reply:
The Example-CarPlay target’s `PRODUCT_MODULE_NAME` build setting is already set to `$(PRODUCT_NAME:c99extidentifier)`, which resolves to `Example_CarPlay`, so `Example_CarPlay.AppDelegate` would be the class name at runtime even without these changes.

Reply:
I'll try that. I prefer not to make these kinds of product name changes if there is an alternative like you mention.

Reply:
@1ec5 - Check out what I did for this. I had to hardcode the product name to "Example" since I can't change the CarPlay target name to Example because of the regular target. I tried forming a variable expression for it but couldn't figure out if that's possible for the setup we want.

Reply:
@1ec5 - Let me know if this seems sufficient. I'd love to finish this off soon.